src/test/java/com/puppycrawl/tools/checkstyle/internal/UtilClassesImmutabilityTest.java

```diff
@@ -0,0 +1,149 @@
+        "byte",
+        "short",
+        "int",
+        "float",
+        "char",
+        "boolean"
+    );
+
+    /**
+     * List of suppressed fields.
+     */
+    private static final Set<String> SUPPRESSED_FIELDS = Set.of(
+        "com.puppycrawl.tools.checkstyle.checks.javadoc.utils.TagInfo.position",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_BIT_SET",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_BYTE_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_DOUBLE_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_INTEGER_OBJECT_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_INT_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_OBJECT_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.CommonUtil.EMPTY_STRING_ARRAY",
+        "com.puppycrawl.tools.checkstyle.utils.JavadocUtil$JavadocTagType.$VALUES",
+        "com.puppycrawl.tools.checkstyle.utils.JavadocUtil$JavadocTagType.ALL",
+        "com.puppycrawl.tools.checkstyle.utils.JavadocUtil$JavadocTagType.BLOCK",
+        "com.puppycrawl.tools.checkstyle.utils.JavadocUtil$JavadocTagType.INLINE",
+        "com.puppycrawl.tools.checkstyle.utils.TokenUtil.TOKEN_IDS",
+        "com.puppycrawl.tools.checkstyle.utils.XpathUtil.TOKEN_TYPES_WITH_TEXT_ATTRIBUTE"
+    );
+
+    /**
+     * Descrie predicate filtering suppressed fields.
+     */
+    private static final DescribedPredicate<JavaField> ARE_NOT_SUPPRESSED =
+        new DescribedPredicate<>("are not suppressed") {
+        @Override
+        public boolean apply(JavaField input) {
+            return !SUPPRESSED_FIELDS.contains(input.getFullName());
+        }
+    };
+
+    /**
+     * ArchCondition for immutable fields.
+     */
+    private static final ArchCondition<JavaField> BE_IMMUTABLE = new ArchCondition<>(
+        "be among default immutable types") {
+        @Override
+        public void check(JavaField item, ConditionEvents events) {
+            final String rawTypeName = item.getRawType().getName();
+            boolean addViolation = false;
+            if (!(PRIMITIVE_TYPES.contains(rawTypeName)
+                || DEFAULT_IMMUTABLE_TYPES.contains(rawTypeName))) {
+                final JavaType javaType = item.getType();
+                addViolation = true;
+                if (javaType instanceof JavaParameterizedType) {
+                    final JavaParameterizedType parameterizedType =
+                        (JavaParameterizedType) javaType;
+                    addViolation = !parameterizedType.getActualTypeArguments().stream()
+                        .allMatch(actualTypeArgument -> {
+                            return DEFAULT_IMMUTABLE_TYPES.contains(
+                                actualTypeArgument.toErasure().getName());
+                        });
+                }
+            }
+            else if ("long".equals(rawTypeName) || "double".equals(rawTypeName)) {
```

Comment:
Why is long/double separated from primitive types? They are immutable if they have `final` otherwise they are not immutable.